sql/core/src/test/java/test/org/apache/spark/sql/sources/v2/JavaSimpleWritableDataSource.java

```diff
@@ -0,0 +1,297 @@
+      assert path != null;
+      try {
+        if (Files.exists(Paths.get(path))) {
+          return new JavaSimpleCSVReadTask(Files.readAllLines(Paths.get(path)).iterator());
+        } else {
+          return new JavaSimpleCSVReadTask(Collections.emptyIterator());
+        }
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    @Override
+    public boolean next() {
+      if (lines.hasNext()) {
+        currentLine = lines.next();
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    @Override
+    public Row get() {
+      String[] values = currentLine.split(",");
+      assert values.length == 2;
+      long l1 = Long.valueOf(values[0]);
+      long l2 = Long.valueOf(values[1]);
+      return new GenericRow(new Object[] {l1, l2});
+    }
+
+    @Override
+    public void close() throws IOException {
+
+    }
+  }
+
+  @Override
+  public DataSourceV2Reader createReader(DataSourceV2Options options) {
+    return new Reader(options.get("path").get());
+  }
+
+
+  class Writer implements DataSourceV2Writer {
+    private String path;
+
+    Writer(String path) {
+      this.path = path;
+    }
+
+    @Override
+    public DataWriteFactory<Row> createWriterFactory() {
+      return new JavaSimpleCSVDataWriteFactory(path);
+    }
+
+    @Override
+    public void commit(WriterCommitMessage[] messages) {
+
+    }
+
+    @Override
+    public void abort() {
+
+    }
+  }
+
+  class InternalRowWriter implements DataSourceV2Writer, SupportsWriteInternalRow {
+    private String path;
+
+    InternalRowWriter(String path) {
+      this.path = path;
+    }
+
+    @Override
+    public DataWriteFactory<InternalRow> createInternalRowWriterFactory() {
+      return new JavaInternalRowCSVDataWriteFactory(path);
+    }
+
+    @Override
+    public void commit(WriterCommitMessage[] messages) {
+
+    }
+
+    @Override
+    public void abort() {
+
+    }
+  }
+
+  static class JavaSimpleCSVDataWriteFactory implements DataWriteFactory<Row>, DataWriter<Row> {
+    private String path;
+    private volatile PrintWriter fileWriter;
+
+    JavaSimpleCSVDataWriteFactory(String path) {
+      this.path = path;
+    }
+
+    JavaSimpleCSVDataWriteFactory(String path, PrintWriter fileWriter) {
+      this.path = path;
+      this.fileWriter = fileWriter;
+    }
+
+    @Override
+    public DataWriter<Row> createWriter(int stageId, int partitionId, int attemptNumber) {
+      try {
+        return new JavaSimpleCSVDataWriteFactory(
+          path, new PrintWriter(new FileWriter(path, true)));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    @Override
+    public void write(Row record) {
+      fileWriter.println(record.getLong(0) + "," + record.getLong(1));
+    }
+
+    @Override
+    public WriterCommitMessage commit() {
+      fileWriter.close();
+      return null;
+    }
+
+    @Override
+    public void abort() {
+      fileWriter.close();
```

Comment:
if you are catching and wrapping exceptions, this should be included in the try/catch clause